### PR TITLE
issue-2-view-entries - Alice Lee

### PR DESCRIPTION
**Pull request by alicejeeunlee**
https://user-images.githubusercontent.com/106103291/179871408-8ed83c85-fe4c-446d-97e4-c8b8f7f52dde.mp4

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -19,4 +19,76 @@ $form.addEventListener('submit', function (event) {
...
+});
+
+var $newButton = document.querySelector('.new-button');
+$newButton.addEventListener('click', function (event) {
+  if (event.target.matches('.new-button')) {
+    $dataViewEntryForm.className = 'container data-form';
+    $dataViewEntries.className = 'container data-entries hidden';
+  }
+  var $formDataView = $dataViewEntryForm.getAttribute('data-view');
+  localStorage.setItem('code-journal-data-view', $formDataView);
```
Comment (codingcodymiller) [inline js/main.js]: Having this separate localStorage item isn't necessary, you should be updating the value of `data.view` every time the view changes, and then you can use the value of `data.view` when the page loads to determine what the last view was that the user was on.

File: js/main.js [hunk 2]
```diff
@@ -19,4 +19,76 @@ $form.addEventListener('submit', function (event) {
...
+    $ul.appendChild(renderEntry(data.entries[i]));
+  }
+  var JSONDataView = localStorage.getItem('code-journal-data-view');
+  if (JSONDataView === 'entries') {
+    $dataViewEntryForm.className = 'container data-form hidden';
+    $dataViewEntries.className = 'container data-entries';
+  } else {
+    $dataViewEntryForm.className = 'container data-form';
+    $dataViewEntries.className = 'container data-entries hidden';
+  }
```
Comment (codingcodymiller) [inline js/main.js]: This is a very manual way of handling view swapping, I suggest you try to implement a system more similar to what you did in the `javascript-view-swapping` exercise that used a loop over all view containers.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (alicejeeunlee) [reply js/main.js]: wowwww totally overlooked this property 😅 next time I have to re-read documentation before starting to code!

**Comment by alicejeeunlee** [comment]
https://user-images.githubusercontent.com/106103291/180034426-fc6b1554-bf6c-4281-aa7c-33f51bda8b5a.mp4

File: js/main.js [hunk 3]
```diff
@@ -56,36 +58,43 @@ window.addEventListener('DOMContentLoaded', function (event) {
...
   if (event.target.matches('.new-button')) {
-    $dataViewEntryForm.className = 'container data-form';
-    $dataViewEntries.className = 'container data-entries hidden';
+    for (var i = 0; i < $allView.length; i++) {
+      if ($allView[i].getAttribute('data-view') === 'entry-form') {
+        $allView[i].className = 'container view';
+      } else {
+        $allView[i].className = 'container view hidden';
+      }
+    }
```
Comment (codingcodymiller) [inline js/main.js]: You've repeated this exact same behavior with only minor adjustments three times. You should make a reusable function that has a parameter of a string which is the name of the view you want to show, which loops over your list of views and shows the view with a matching name and hides all other views that don't match.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 3 above)
Comment (alicejeeunlee) [reply js/main.js]: 🤔

**Comment by alicejeeunlee** [comment]
https://user-images.githubusercontent.com/106103291/180044376-89ba8066-9309-4f01-acc4-4c64406c478b.mp4

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**